blog/2020/12/precompile_tutorial.md

```diff
@@ -0,0 +1,586 @@
+@def title = "Tutorial on precompilation"
+@def authors = "Tim Holy"
+@def published = "28 December 2020"
+@def rss_pubdate = Date(2020, 12, 28)
+@def rss = """Tutorial on precompilation"""
+
+~~~
+  <link rel="stylesheet" type="text/css" href="/assets/blog/2020-precompile/boxes.css" />
+~~~
+
+One of the main foci of development during Julia 1.6 has been to reduce *latency*, the delay between starting your session and getting useful work done.  This is sometimes called "time to first plot," although it applies to far more than just plotting.
+While a lot of work (and success) has gone into reducing latency in Julia 1.6, users and developers will naturally want even more.  This is the inaugural post in a short series devoted to the topic of what package developers can do to reduce latency for their users.  This particular installment covers background material that will hopefully be useful in later installments.
+
+<div class="note"></div>
+**NOTE**: Fixme!
+
+## Sources of latency, and reducing it with `precompile`
+
+Most of Julia's latency is due to *code loading* and *compilation*. Julia's dynamic nature also makes it vulnerable to *invalidation* and the subsequent need to recompile previously-compiled code; this topic has been covered in a previous [blog post][invalidation], and that material will not be rehashed here.  In this series, it is assumed that invalidations are not a dominant source of latency.  (You do not need to read the previous blog post to understand this one.)
+
+In very rough terms, `using SomePkg` loads the method definitions, after which calling `SomePkg.f(args...)` forces `SomePkg.f` to be compiled (if it hasn't been already) for the specific types in `args...`.  The primary focus of this series is to explore the opportunity to reduce the cost of compilation.  We'll focus a lot on *precompilation*,
+
+```
+julia> using SomePkg
+[ Info: Precompiling SomePkg [12345678-abcd-9876-efab-1234abcd5e6f]
+```
```

Comment:
worth adding something like this?

---
or as of 1.6, at the end of `pkg>` actions
```
(@1.6) pkg> add SomePkg
    Updating registry at `~/.julia/registries/General`
   Resolving package versions...
    Updating `~/.julia/environments/v1.6/Project.toml`
  [12345678] + SomePkg v0.1.0
    Updating `~/.julia/environments/v1.6/Manifest.toml`
  [12345678] + SomePkg v0.1.0
  [abcdefgh] + SomeDep v0.1.0
 Precompilng project...
  Progress [========================================>]  2/2
2 dependencies successfully precompiled in 3 seconds
```

Reply:
I added a brief version of that.